Fix undefined error for `ActionController::Parameters`

maclover7 opened the pull request:
This was introduced as apart of the 9-CVE-push (aka 5.0.0.beta1.1).

r? @tenderlove

maclover7 committed:
Fix undefined error for `ActionController::Parameters`

maclover7 commented:
This will also need to be backported to 4-2-stable.

prathamesh-sonpatki commented on actionpack/lib/abstract_controller/rendering.rb:
```diff
@@ -82,13 +82,10 @@ def view_assigns
-        unless action.permitted?
-          raise ArgumentError, "render parameters are not permitted"
-        end
+      if action.respond_to?(:permitted?)
```
Doesn't this change behavior? We are now raising if action respods to `permitted?` not *if* action is permitted or not?

prathamesh-sonpatki commented:
Can this undefined error be not fixed by just adding `require 'action_controller/metal/strong_parameters'` in this file?

vipulnsward commented:
This worked for me- https://gist.github.com/vipulnsward/3fc7cf7c51d60f6c4e6c
@prathamesh-sonpatki I think the purpose would be to keep AC isolated from Metal.

maclover7 commented:
@prathamesh-sonpatki - we are trying to avoid coupling on AbstractController to ActionController -- it is meant to be the other way around. Yes a require would fix the issue, but we also need to remove the underlying coupling that's going on here.

maclover7 committed:
Fix sanitizer tests

These tests were failing due to backwards incompatible changes, as apart
of the v1.0.3 release of rails-html-sanitizer.
